### ISSUE TITLE
Inclination vs elevation

### ISSUE THREAD

**Issue by Lestropie** [MRtrix3/mrtrix3#2630]
I've been aware of this one for a while, but didn't realise that I had not listed it. Has come back to the front of my mind due to playing with particularly #2626.

[Spherical coordinates](https://en.wikipedia.org/wiki/Spherical_coordinate_system) can be defined using different conventions, in terms or presence or absence of elements / ordering of elements / symbols for elements / reference & range for quantities. In BIDS [BEP016](https://github.com/bids-standard/bids-bep016) I've [stuck to ISO standard](https://github.com/bids-standard/bids-bep016/blob/8dee2f0f75f8c2dab1c6916d9f2f277e7101ca43/src/derivatives/05-diffusion-derivatives.md?plain=1#L147-L185) as much as possible. In *MRtrix3* code, one frequently sees references in this context to "az_el", ie. azimuth-elevation pairs. However the [`cos`](https://github.com/MRtrix3/mrtrix3/blob/3.0.4/core/math/sphere.h#L45) / [`acos`](https://github.com/MRtrix3/mrtrix3/blob/3.0.4/core/math/sphere.h#L87) transformations indicate that the origin of the second spherical angle is the z-axis. This would make it an *inclination* rather than elevation angle: the former uses the angle from the zenith direction, the latter uses the angle from the reference plane.

- [ ] Do code audit to make sure that there are no fundamental conflicts in how such values are transformed / interpreted in various places.
    One potential culprit I'm thinking of here is the calculation of lighting effects in OpenGL code.
- [ ] Find any manual manipulation of spherical vs cartesian directions and centralise it all to `MR::Math::Sphere`.
    I've already done this a little in #2626 and related, but it needs a more exhaustive search.
- [ ] Rename variables to properly reflect the actual encoding.
- [ ] Add somewhere in documentation the way in which such data are imported / exported.
    This could potentially include a copy-paste help page Description paragraph, or a version-matched documentation hyperlink.

**Comment by Lestropie** [comment]
True; might we worth expanding the scope of the issue to cover the order of coefficients. The expected ranges should ideally not be of consequence since they should wash out in any relevant calculations; what matters most is the definition of where zero is. For inclination vs elevation as polar angle in particular, there should be a bunch of `cos()` / `acos()` vs. `sin()` / `asin()` calls in conversions to / from Cartesian.

The inhomogeneity certainly justifies my assertion of enforcing ISO in BEP016... 😄

**Comment by fionaEyoung** [comment]
> The expected ranges should ideally not be of consequence since they should wash out in any relevant calculations

Indeed, and I was thinking that some functions may enforce the ranges through input checks, but for example even though the scipy `sph_harm` [documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.sph_harm.html#scipy.special.sph_harm) says "must be in `[0, 2*pi]`", it looks like that isn't actually inforced and it accepts negative values just fine...

**Comment by Lestropie** [comment]
In #2918 I have made some changes in this respect.

From looking at the code I'm reasonably confident that in the contexts of dixel set handling and SH, the polar angle is consistently interpreted as an *inclination*, not an *elevation*, contrary to variable naming.

I have however reverted `GUI::LightingDock` to persist with "elevation". Based on the ODF renderer (since it predates the tractogram lighting and is therefore a more suitable reference), I believe that interpretation of the slider position as an "elevation" makes more sense. Hopefully the screenshots below show this. It is however the case that the interpretation of the light position in the tractography lighting is the exact opposite to that of the ODF renderer. In those screenshots you can see that I've put the azimuth and elevation sliders in the same positions between the ODF and tractography tools, but the lighting is always incongruent.

<img width="1920" height="1053" alt="Image" src="https://github.com/user-attachments/assets/171fde8b-680e-41c6-92b1-3444f8cf4e6d" />

<img width="1920" height="1053" alt="Image" src="https://github.com/user-attachments/assets/d8d358dd-9a5a-4599-a0ba-4b521fa2ba6a" />

<img width="1920" height="1053" alt="Image" src="https://github.com/user-attachments/assets/9d444277-abb2-4602-be3e-7c0c83991c32" />

<img width="1920" height="1053" alt="Image" src="https://github.com/user-attachments/assets/3787c5f0-309a-4570-81b2-68ec743c3a28" />
